SConstruct

```diff
@@ -932,20 +962,16 @@ if env['system_eigen'] in ('n', 'default'):
                          'Try manually checking out the submodule with:\n\n'
                          '    git submodule update --init --recursive ext/eigen\n')
 
+if env['system_eigen']:
+    eigen_include = ['<eigen3/Eigen/Core>', '<eigen3/Eigen/src/Core/util/Macros.h>']
+else:
+    eigen_include = ['"../ext/eigen/Eigen/Core"', '"../ext/eigen/Eigen/src/Core/util/Macros.h"']
+eigen_version = []
+for ver in ['EIGEN_WORLD_VERSION', 'EIGEN_MAJOR_VERSION', 'EIGEN_MINOR_VERSION']:
+    eigen_version.append(conf.TryRun(get_expression_value(eigen_include, ver), '.cpp')[1].strip())
 
-def get_expression_value(includes, expression):
-    s = ['#include ' + i for i in includes]
-    s.extend(('#define Q(x) #x',
-              '#define QUOTE(x) Q(x)',
-              '#include <iostream>',
-              '#ifndef SUNDIALS_PACKAGE_VERSION', # name change in Sundials >= 3.0
-              '#define SUNDIALS_PACKAGE_VERSION SUNDIALS_VERSION',
-              '#endif',
-              'int main(int argc, char** argv) {',
-              '    std::cout << %s << std::endl;' % expression,
-              '    return 0;',
-              '}\n'))
-    return '\n'.join(s)
+env['EIGEN_VERSION'] = '.'.join(eigen_version)
```

Comment:
You should be able to get all of this info in a single call to `get_expression_value`. The expression should be something like
```cpp
QUOTE(EIGEN_WORLD_VERSION) "." QUOTE(EIGEN_MAJOR_VERSION) "." QUOTE(EIGEN_MINOR_VERSION)
```
where the `QUOTE` macro is already defined in the `get_expression_value` program.